tests/snmp/test_snmp_phy_entity.py

```diff
@@ -349,60 +484,293 @@ def test_fan_info(duthosts, enum_rand_one_per_hwsku_hostname, snmp_physical_enti
             else:
                 parent_oid = MODULE_TYPE_FAN_DRAWER + parent_position * MODULE_INDEX_MULTIPLE
         expect_oid = parent_oid + DEVICE_TYPE_FAN + position * DEVICE_INDEX_MULTIPLE
-        assert expect_oid in snmp_physical_entity_info, 'Cannot find fan {} in physical entity mib'.format(
-            name)
+        assert expect_oid in snmp_physical_entity_info, (
+            "Cannot find fan {} in physical entity mib. "
+            "Redis DB key: {}. Redis DB value: {}. "
+        ).format(
+            name,
+            key,
+            fan_info
+        )
         fan_snmp_fact = snmp_physical_entity_info[expect_oid]
-        assert fan_snmp_fact['entPhysDescr'] == name
-        assert fan_snmp_fact['entPhysContainedIn'] == CHASSIS_SUB_ID if parent_name == CHASSIS_KEY else parent_oid
-        assert fan_snmp_fact['entPhysClass'] == PHYSICAL_CLASS_FAN
-        assert fan_snmp_fact['entPhyParentRelPos'] == position
-        assert fan_snmp_fact['entPhysName'] == name
-        assert fan_snmp_fact['entPhysHwVer'] == ''
-        assert fan_snmp_fact['entPhysFwVer'] == ''
-        assert fan_snmp_fact['entPhysSwVer'] == ''
-        assert fan_snmp_fact['entPhysSerialNum'] == '' if is_null_str(fan_info['serial']) else fan_info[
-            'serial']
-        assert fan_snmp_fact['entPhysMfgName'] == ''
+        assert fan_snmp_fact['entPhysDescr'] == name, (
+            "Fan description mismatch. Expected '{}', but got '{}'. "
+            "Redis DB key: {}. Redis DB value: {}. "
+        ).format(
+            name,
```

Comment:
The rule is what variable in assert statement, just print them out, don't print out other variables not in statement.